nums/core/array/blockarray.py

```diff
@@ -160,6 +160,30 @@ def squeeze(self):
             new_block_shape.append(b)
         return self.reshape(new_shape, new_block_shape)
 
+    def swapaxes(self, axis1, axis2):
+        meta_swap = self.grid.to_meta()
+        shape = list(meta_swap["shape"])
+        block_shape = list(meta_swap["block_shape"])
+        dim = len(shape)
+        if axis1 >= dim or axis2 >= dim:
+            raise ValueError("axis is larger than the array dimension")
+        shape[axis1], shape[axis2] = shape[axis2], shape[axis1]
+        block_shape[axis1], block_shape[axis2] = block_shape[axis2], block_shape[axis1]
+        meta_swap["shape"] = tuple(shape)
+        meta_swap["block_shape"] = tuple(block_shape)
+        grid_swap = ArrayGrid.from_meta(meta_swap)
+        rarr_swap = BlockArray(grid_swap, self.system)
+        rarr_src = self.blocks.swapaxes(axis1, axis2).reshape(-1)
+        rarr_tgt = rarr_swap.blocks.reshape(-1)
+        for idx, j in np.ndenumerate(rarr_src):
+            rarr_tgt[idx] = j.copy()
+
+        for grid_entry in rarr_swap.grid.get_entry_iterator():
```

Comment:
Since `swapaxes` is no longer an in-place operation, please update this loop to assign the returned block accordingly.